src/elements/content-sidebar/__tests__/useSidebarMetadataFetcher.test.tsx

```diff
@@ -106,4 +144,46 @@ describe('useSidebarMetadataFetcher', () => {
             }),
         );
     });
+
+    test('should handle metadata instance creation', async () => {
+        mockAPI.getMetadata.mockImplementation((file, successCallback) => {
+            successCallback({ templateInstances: mockTemplateInstances, templates: mockTemplates });
+        });
+        mockAPI.createMetadata.mockImplementation((file, template, successCallback) => {
+            successCallback(newTemplateInstance);
+        });
+
+        const { result } = setupHook();
+
+        await waitFor(() => expect(result.current.templateInstances).toEqual(mockTemplateInstances));
```

Comment:
I mention this in another PR, we dont need the `await waitFor` on any of the `expects`. if you are doing something like this ` await waitFor(() => result.current.createMetadataInstance(newTemplateInstance));` then you still need to do that

Reply:
thank you, changed